rom/dev/src/main.rs

```diff
@@ -415,3 +427,10 @@ fn validate_trng_config(env: &mut RomEnv) {
         matches!(env.trng, Trng::MfgMode()),
     );
 }
+
+/// Check if the hardware version is supported
+/// Supports 2.0.2+ and 2.1.0+
+fn is_supported_hw_version(cptra_gen: &CptraGeneration) -> bool {
+    !(cptra_gen.major_version() != 2
+        || (cptra_gen.minor_version() == 0 && cptra_gen.patch_version() < 2))
```

Comment:
This expression is difficult to reason about.

Maybe a match statement is clearer? Or using `&&` instead of `||`?

Example:

```rust
match (cptra_gen.major_version(), cptra_gen.minor_version(), cptra_gen.patch_version()) {
  (2, 0, 0..=1) => false,
  _ => true,
}
```